doc/iterations/2024.5/shapes/configurable_visibility.adoc

```diff
@@ -0,0 +1,47 @@
+= Configurable visibility
+
+== Problem
+
+Sirius Web doesn't provide a way for specifiers to set the default visibility of a node (hidden/visible). 
+It also doesn't provide a way to let specifiers define tools that hide/reveal nodes as part of other operations (e.g. to reveal a hidden node when a child node is created).
+Nodes are always created visible, and the only way to hide them right now is that the end user does it manually.
+
+== Key Result
+
+- Specifiers must be able to define NodeDescriptions that are hidden/visible by default.
+- Specifiers must be able to call hide/reveal services in their tool to hide/reveal a set of nodes.
+
+== Solution
+
+A new field _Is Hidden By Default Expression_ should be displayed in NodeDescription. 
+The field contains an boolean expression which is evaluated on the first rendering of the node to set its initial hidden/visible state.
+If the expression is empty it is evaluated to `false`, which means that the node is visible by default (this preserves Sirius Web's default behavior to show new nodes by default). 
+
+The `diagramServices` variable should contain new services to hide/reveal nodes.
+
+== Scenario
+
+1. Default Visibility State
+- A specifier adds a NodeDescription in the view DSL
+- The _Is Hidden By Default Expression_ field is available and its default value is empty, which means visible
```

Comment:
This is where I agree, we need something simple and `isHiddenByDefaultExpression` asking for a boolean is simple. I just think we need a `isFadedByDefaultExpression` asking also for a boolean and if both are true, hidden wins. We can't ask specifiers to return an enumeration literal.

Reply:
done